service/matching/handler.go

```diff
@@ -143,12 +143,24 @@ func (h *Handler) PollForDecisionTask(ctx context.Context,
 // QueryWorkflow queries a given workflow synchronously and return the query result.
 func (h *Handler) QueryWorkflow(ctx context.Context,
 	queryRequest *m.QueryWorkflowRequest) (*gen.QueryWorkflowResponse, error) {
-	return nil, &gen.InternalServiceError{Message: "Not implemented yet"}
+	scope := metrics.MatchingQueryWorkflowScope
+	sw := h.startRequestProfile("QueryWorkflow", scope)
+	defer sw.Stop()
+
+	response, error := h.engine.QueryWorkflow(ctx, queryRequest)
+	h.Service.GetLogger().Debug("Engine returned from QueryWorkflow")
+	return response, h.handleErr(error, scope)
 }
 
 // RespondQueryTaskCompleted responds a query task completed
 func (h *Handler) RespondQueryTaskCompleted(ctx context.Context, request *m.RespondQueryTaskCompletedRequest) error {
-	return &gen.InternalServiceError{Message: "Not implemented yet"}
+	scope := metrics.MatchingRespondQueryTaskCompletedScope
+	sw := h.startRequestProfile("RespondQueryTaskCompleted", scope)
+	defer sw.Stop()
+
+	error := h.engine.RespondQueryTaskCompleted(ctx, request)
+	h.Service.GetLogger().Debug("Engine returned from QueryWorkflow")
```

Comment:
incorrect log message, is this log useful even as a debug ?

Reply:
this was not intended, should be removed.